packages/block-editor/src/components/use-block-sync/index.js

```diff
@@ -174,9 +204,26 @@ export default function useBlockSync( {
 				const updateParent = isPersistent
 					? onChangeRef.current
 					: onInputRef.current;
-				updateParent( blocks, {
-					selectionStart: getSelectionStart(),
-					selectionEnd: getSelectionEnd(),
+
+				const outgoingBlocks = shouldMapBlockIds()
+					? mapBlocks( blocks, instanceId, DIRECTION_OUT )
+					: blocks;
+
+				updateParent( outgoingBlocks, {
+					selectionStart: shouldMapBlockIds()
```

Comment:
You use this same conditional again when calculating `selectionEnd`. Why not just have two objects and do something like this:

```js
updateParent(
	outgoingBlocks,
	shouldMapBlockIds()
		? {
			selectionStart: mapBlockId(
				getSelectionStart(),
				instanceId,
				DIRECTION_OUT
			),
			selectionEnd: mapBlockId(
				getSelectionEnd(),
				instanceId,
				DIRECTION_OUT
			)
		  }
		: {
			selectionStart: getSelectionStart(),
			selectionEnd: getSelectionEnd()
		  }
)
```

I think that would be slightly better for performance.

Reply:
7cfc4ac